api/package.json

```diff
@@ -98,7 +97,7 @@
     "nyc": "15.1.0",
     "sinon": "15.1.2",
     "ts-loader": "9.5.1",
-    "typescript": "4.4.4",
+    "typescript": "5.6.3",
```

Comment:
@open-telemetry/javascript-maintainers what do you think about making this change in the API package? I think we should probably just go for it:

1. We can't stay on 4.4.4 forever
2. We don't want to go to API 2.0 any time in the foreseeable future

I think for both of these to be true, we have to eventually make this change in a minor API version, painful though it may be to some small number of users.

In order to not make this change, we'd have to keep a different version of typescript here than everywhere else. IDK how painful this would be in practice in our monorepo (maybe not that bad?)

Reply:
Hmm, I think at least typedoc is coupled with the typescript version - so it may hold us back with quite a few dependencies as well. I'm for updating. Maybe we could do the following: 
- we compile using `typescript@5.6.3` 
- we add a integration test package (not linked to the monorepo) that installs the local `@opentelemetry/api` compiled with `5.6.3`, and that checks if current features compile with TypeScript `4.4.4` and only runs in the test workflow. Maybe it even works fine since we're not using any new TypeScript features. 🤔 
  - If it does work fine the question might become: do we want to allow new API features that use new TypeScript features that would not work with 4.4.4 in minor versions?

Here's me hoping that bumping the API major will be allowed at some point in the future.